Data-Structures/Heap/KeyPriorityQueue.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * KeyPriorityQueue is a priority queue based on a Minimum Binary Heap.
+ *
+ * Minimum Binary Heaps are binary trees which are filled level by level
+ * and then from left to right inside a depth level.
+ * Their main property is that any parent node has a smaller or equal priority to all of its children,
+ * hence the root of the tree always has the smallest priority of all nodes.
+ *
+ * This implementation of the Minimum Binary Heap allows for nodes to contain both a key and a priority.
+ * WARNING: keys must be Integers as they are used as array indices.
+ *
+ * In this implementation, the heap is represented by an array with nodes ordered
+ * from root-to-leaf, left-to-right.
+ * Therefore, the parent-child node relationship is such that
+ *      * the children nodes positions relative to their parent are: (parentPos * 2 + 1) and (parentPos * 2 + 2)
+ *      * the parent node position relative to either of its children is: Math.floor((childPos - 1) / 2)
+ *
+ * More information and visuals on Binary Heaps can be found here: https://www.geeksforgeeks.org/binary-heap/
+ */
+
+// Priority Queue Helper functions
+function getParentPosition (position) {
+  // Get the parent node of the current node
+  return Math.floor((position - 1) / 2)
+}
+function getChildrenPosition (position) {
+  // Get the children nodes of the current node
+  return [2 * position + 1, 2 * position + 2]
+}
+
+class KeyPriorityQueue {
+  // Priority Queue class using Minimum Binary Heap
+  constructor () {
+    this._heap = []
+    this.keys = {}
```

Comment:
Why not use a ES6 `Map` here?

Reply:
Indeed it helps clarify the code.
I made the change.